pkg/nfd-master/nfd-master.go

```diff
@@ -605,26 +635,38 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, nfdAnnotations, labels map[string]string, features *nfdv1alpha1.Features) error {
 	if labels == nil {
 		labels = make(map[string]string)
 	}
 
-	crLabels, crTaints := m.processNodeFeatureRule(features)
+	if nfdAnnotations == nil {
+		nfdAnnotations = make(map[string]string)
+	}
+
+	featureAnnotations := make(map[string]string)
+
+	crLabels, crAnnotations, crTaints := m.processNodeFeatureRule(features)
 
 	// Mix in CR-originated labels
 	for k, v := range crLabels {
 		labels[k] = v
 	}
 
+	for k, v := range crAnnotations {
+		featureAnnotations[k] = v
+	}
+
 	labels, extendedResources := m.filterFeatureLabels(labels)
 
+	featureAnnotations = m.filterFeatureAnnotations(featureAnnotations)
```

Comment:
We could simplify a bit. No need to declare and copy the map above

```suggestion
	featureAnnotations := m.filterFeatureAnnotations(crAnnotations)
```